interview-questions.md

```diff
@@ -6,38 +6,50 @@ First, without external resources. Challenge yourself to answer from memory.
 
-Your answer:
+Your answer:go back and add the foreign edit it so it has the foreign key. The foreign key is the id of the colums in the second table. it would be on the student model
 
-Researched answer:
+Researched answer: a foreign key is a column of the table which belongs to the other one. Using migrate can help change or add a column if the developer forgot to add one. The name of the foreign key is the name of the column, it would be the student model since it belongs to the cohort table.
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: the ones that invlolve seeing, creating, updating and deleting them so the computer and/or software knows what to see, change, create, update and delete. this would be view, put, post , delete, and patch.
 
-Researched answer:
+Researched answer: I couldnt find any online sources that informed me of the required routes that needed params
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: rails g model, this will create a table for rails that can be fillable with different instances. rails g resources, does similair function to rails g model but removes view by default for backend developing. i cant think of any others at the top of my head
 
-Researched answer:
+Researched answer: rails generate -h would give a list of available generate commands to use. -h is listed as the help option. there is rails generate - f which forces the terminal to write over past saved files. rails g -q will suppress status output 
+source: https://medium.com/@josephdlawson21/using-the-generate-command-in-rails-9c738380f2d9
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET" location: /students
+action: "GET" location: /students   index.controller it would show list of students
 
-action: "POST" location: /students
+action: "POST" location: /students update.controller it would allow creation of students info
 
-action: "GET" location: /students/new
+action: "GET" location: /students/new index.controller 
 
-action: "GET" location: /students/2
+action: "GET" location: /students/2 index.controller looks at one of the students info
 
-action: "GET" location: /students/2/edit
+action: "GET" location: /students/2/edit update.controller allow edit or update of student information
 
-action: "PATCH" location: /students/2
+action: "PATCH" location: /students/2 update.controller allow edit or update of student information
 
-action: "DELETE" location: /students/2
+action: "DELETE" location: /students/2 delete.controller would delete information of students
```

Comment:
The index controller and update controller would only relate to one of these Rails routes.  You can find this in the RESTful routes section in the syllabus.